enhancements/local-storage/automatic-detection.md

```diff
@@ -64,40 +65,299 @@ and we can use that information to sort devices into storage classes based on th
+  - Update the `status` of the `LocalVolumeDiscoveryResult` with `discoveredTimeStamp`. This represents the timestamp when the discovery was completed.
+`
+#### `Auto Device Discovery` Status
+- The `LocalVolumeDiscovery` controller will track the status of the disovery process by:
+  - Tracking the `LocalVolumeDiscoveryResult` CR for each node. For `n` nodes, the controller should ensure that `n` `LocalVolumeDiscoveryResult` CR should be created.
+  - Once all the `LocalVolumeDiscoveryResult` CR are created, the controller would check `discoveredTimeStamp` timestamp for each `LocalVolumeDiscoveryResult` CR.
+  - When the `discoveredTimeStamp` timestamp for each `LocalVolumeDisocveryResult`CR is greater than `discoveryStarted` timestamp, then update the `Phase` in the `LocalVolumeDiscovery` CR to `Discovered`.
+- Controller should update the `LocalVolumeDiscovery` `Phase` as `Error` if:
+  - There is some error in running the `DaemonSet`
+  - The`LocalVolumeDiscoveryResult` CR is not created for each valid nodes even after a fixed interval time.
+  - `discoveredTimeStamp` is not available in one of more `LocalVolumeDiscoveryResult` CR.
+
+#### `Auto Device Discovery` Cadence
+- Auto discovery of devices should be a continuous proccess to support on-going management of devices and to display the devices in an inventory list.
+- Auto discovery should continuously check for the list of available devices on a given node after every fixed interval of time to identify if a new device has been added or an existing device has been removed.
+- In case of any addtion or deletion of devices, the `LocalVolumeDiscoveryResult` CR should be updated to reflect the new set of devices on the node.
+- Continuous discovery can be triggered based following criteria:
+  - After every fixed interval of time.
+  - After any `udev` events on the block device (like `add` or `remove`)
+
+#### `Auto Device Discovery` Node Deletion
+- User updates the `LocalVolumeDiscovery` CR to remove node(s) for discovery:
+  - Daemonset would be updated with the new set of nodes, by the controller.
+  - This would automatically delete the daemonset pod running on the removed node(s).
+  - Controller would delete the `LocalVolumeDiscoveryResult` CR for the removed Node. (Can the `LocalVolumeDiscoveryResult` CR be garbage collected?)
+
+#### `Auto Device Discovery` Node Addition
+- User updates the `LocalVolumeDiscovery` CR to track new node(s) for discovery:
+  - DaemonSet would be updated with the new set of nodes, by the controller.
+  - This would automatically create a daemonset pod and start discovery on the newly added node(s).
+
+#### `Auto Device Discovery` CR Deletion
+- A delete request on the `LocalVolumeDiscovery` CR would trigger a graceful deletion with help of a finalizer and it would:
+  - Delete the `LocalVolumeDiscoveryResult` CR for each node. (Can the `LocalVolumeDiscoveryResult` CR be garbage collected?)
```

Comment:
I'm not sure we should delete the discovery results when the discovery CR is deleted. Maybe the admin wants to stop the discovery process from running, but still analyze the discovery results. So the admin could choose separately when to delete the result CRs.

Reply:
Thanks for the comments. I'll update the design not to delete the discovery results after discovery CR is deleted.